### PR TITLE
feat: added gliding between menu items

### PR DESCRIPTION
**Pull request by iSa1vatore**

### CODE REVIEW

File: lib/src/utils/menu_config.dart [hunk 1]
```diff
@@ -68,17 +69,22 @@ class MenuConfig extends InheritedWidget {
...
+    required this.glideState,
   });
 
   /// Whether the pull-down menu has any [PullDownMenuItem]s with leading
   /// widget such as chevron.
   final bool hasLeading;
 
+  ///
+  // TODO(salvatore): need doc
+  final ValueNotifier<MenuGlideState> glideState;
```
Comment (notDmDrl) [inline lib/src/utils/menu_config.dart:81]: Will it be better to make this ValueNotifer global somewhere in glide_state.dart?

Similar to https://github.com/notDmDrl/pull_down_button/blob/main/lib/src/utils/route.dart#L144

And then just use ValueListenableBuilder directly in MenuActionGestureDetector

Comment (iSa1vatore) [reply lib/src/utils/menu_config.dart]: I don't really like global variables, I think we can put glideState and menuAlignmentNotifier in PullDownMenuRoute with no problem

I added a new commit, it's cleaner in my opinion

What do you think? :)

Comment (notDmDrl) [reply lib/src/utils/menu_config.dart]: Looked at new commit, indeed looks cleaner 👍

**Comment by iSa1vatore** [comment]
We will probably have to use the term "continuous swipe," which is the official name. (I guess.)

In the code, we will also need to replace "Glide" with "ContinuousSwipe".

There is also the problem that this "continuous swipe" does not work when opening, you have to first let go of your finger and then drive through the menu.

At the moment I do not know exactly how to solve this problem, I will try to fix it.


https://user-images.githubusercontent.com/42691184/225150387-a1af1916-ed87-41e1-959e-e533cdc17960.mov

**Comment by notDmDrl** [comment]
> We will probably have to use the term "continuous swipe," which is the official name. (I guess.)

I'll try to search for some info about that too.

> In the code, we will also need to replace "Glide" with "ContinuousSwipe".

SGTM

> At the moment I do not know exactly how to solve this problem, I will try to fix it.

We can merge it without this and add a fix later, if you want.

File: lib/src/_internals/gesture_detector.dart [hunk 2]
```diff
@@ -45,6 +47,46 @@ class _MenuActionGestureDetectorState extends State<MenuActionGestureDetector> {
 
   late final enabled = widget.onTap != null;
 
+  Offset get _currentPosition {
```
Comment (notDmDrl) [inline lib/src/_internals/gesture_detector.dart:50]:
```suggestion
    Offset get _currentPosition => (context.findRenderObject() as RenderBox).localToGlobal(Offset.zero);
```

Comment (notDmDrl) [reply lib/src/_internals/gesture_detector.dart:50]: and _currentSize below

**Review by notDmDrl — COMMENTED** [review]
I assume it's ready for review

LGTM with small style nits

**Pull request merged**